atoma-sui/src/client.rs

```diff
@@ -0,0 +1,1023 @@
+    ///
+    /// This function queries the Sui blockchain to find a NodeBadge object owned by the specified
+    /// address that was created by the specified package. The NodeBadge represents a node's
+    /// registration in the Atoma network.
+    ///
+    /// # Arguments
+    ///
+    /// * `client` - A reference to the SuiClient used to interact with the blockchain
+    /// * `package` - The ObjectID of the Atoma package that created the NodeBadge
+    /// * `active_address` - The SuiAddress to query for owned NodeBadge objects
+    ///
+    /// # Returns
+    ///
+    /// Returns `Option<(ObjectID, u64)>` where:
+    /// - `Some((object_id, small_id))` if a NodeBadge is found, where:
+    ///   - `object_id` is the unique identifier of the NodeBadge object
+    ///   - `small_id` is the node's numeric identifier in the Atoma network
+    /// - `None` if:
+    ///   - No NodeBadge is found
+    ///   - The query fails
+    ///   - The object data cannot be parsed
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// use sui_sdk::SuiClient;
+    /// use sui_sdk::types::base_types::{ObjectID, SuiAddress};
+    ///
+    /// async fn example(client: &SuiClient) {
+    ///     let package_id = ObjectID::new([1; 32]);
+    ///     let address = SuiAddress::random_for_testing_only();
+    ///     
+    ///     match get_node_badge(client, package_id, address).await {
+    ///         Some((badge_id, small_id)) => {
+    ///             println!("Found NodeBadge: ID={}, small_id={}", badge_id, small_id);
+    ///         }
+    ///         None => {
+    ///             println!("No NodeBadge found for address");
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// # Implementation Notes
+    ///
+    /// - The function queries up to 100 objects at a time
+    /// - The function filters objects by package and looks for the specific NodeBadge type
+    /// - Object content is parsed to extract the small_id from the Move object's fields
+    pub(crate) async fn get_node_badge(
+        client: &SuiClient,
+        package: ObjectID,
+        active_address: SuiAddress,
+    ) -> Option<(ObjectID, u64)> {
+        let mut cursor = None;
+        loop {
+            let Page {
+                data,
+                has_next_page,
+                next_cursor,
+            } = match client
+                .read_api()
+                .get_owned_objects(
+                    active_address,
+                    Some(SuiObjectResponseQuery {
+                        filter: Some(SuiObjectDataFilter::Package(package)),
+                        options: Some(SuiObjectDataOptions {
+                            show_type: true,
+                            show_content: true,
+                            ..Default::default()
+                        }),
+                    }),
+                    cursor,
+                    Some(100),
```

Comment:
can we move this constant, to some constant at the top of the file, like `PAGE_SIZE` or something like that.

Reply:
Done